System/SpatialPartitionGrid.h

```diff
@@ -50,54 +45,72 @@ namespace RTE {
 		const std::vector<MovableObject *> & GetMOsInRadius(const Vector &centre, float radius, int ignoreTeam) const;
 
 		/// <summary>
-		/// Get MOIDs that are potentially overlapping a pixel
+		/// Gets the MOIDs that are potentially overlapping the given x and y Scene coordinates.
 		/// </summary>
-		const SpatialPartitionGrid::MOIDList & GetMOIDsAtPosition(int x, int y, int ignoreTeam) const;
+		/// <param name="x">The x coordinate to check.</param>
+		/// <param name="y">The y coordinate to check.</param>
+		/// <param name="ignoreTeam">The team to ignore when getting MOIDs.</param>
+		/// <returns>A vector of MOIDs that are potentially overlapping the x and y coordinates.</returns>
+		const std::vector<int> & GetMOIDsAtPosition(int x, int y, int ignoreTeam) const;
 #pragma endregion
 
 	private:
+
+		/// <summary>
+		/// Gets the Id of the cell at the given SpatialPartitionGrid coordinates, automatically accounting for wrapping.
+		/// </summary>
+		/// <param name="cellX">The x coordinate of the cell to get the Id of.</param>
+		/// <param name="cellY">The y coordinate of the cell to get the Id of.</param>
+		/// <returns>The Id of the cell at the given SpatialPartitionGrid coordinates.</returns>
 		int GetCellIdForCellCoords(int cellX, int cellY) const;
 
-		int m_Width; // Grid width in cells
-		int m_Height; // Grid height in cells
-		int m_CellSize; // Cell size, in pixels
+		int m_Width; //!< The width of the SpatialPartitionGrid, in cells.
+		int m_Height; //!< The height of the SpatialPartitionGrid, in cells.
+		int m_CellSize; //!< The size of each of the SpatialPartitionGrid's cells, in pixels.
 
-		// We store a list per team, so overlapping actors don't waste loads of time collision checking against themselves
-		// Note, this is this list of moids that are potentially colliding per team, not the list of MOIDS per team
-		// So the list for team 1 includes the MOIDs for team 2, 3, 4, and no-team
-		typedef std::array<std::vector<MOIDList>, Activity::MaxTeamCount+1> CellsPerTeam;
-		CellsPerTeam m_Cells;
+		// We store a list per team, so overlapping Actors don't waste loads of time collision checking against themselves.
+		// Note that this is this list of MOIDs that are potentially colliding per team, so the list for team 1 contains the MOIDs for team 2, 3, 4, and no-team, as well as anything for team 1 that doesn't ignore team hits.
+		std::array<std::vector<std::vector<int>>, Activity::MaxTeamCount + 1> m_Cells; //!< Array of cells for each team. The outside-vector is the vector of cells for the team, and each inside-vector entry contains all MOIDs in the cell's space that can collide with that team.
 
-		// We also don't want to waste loads of time looping through and unclearing unused cells, so store a set of used cells
-		std::unordered_set<int> m_UsedCells;
+		std::unordered_set<int> m_UsedCellIds; //!< Set of used cell Ids, maintained to avoid wasting time looping through and clearing unused cells.
```

Comment:
Hitherto, you changed Ids to IDs for consistencies sake. But I prefer this anyways.

Reply:
I know, and I wondered if you'd call me on that haha.
I'm actually very much against ID, because it's not an acronym, and I figure this is a brand new class and can get away with a sneaky change :D

If we're both okay with it, we can just keep it a secret haha

Reply:
Sounds good to me.